### DISCUSSION TITLE
Custom raster map markers from file on local or remote disk

### DISCUSSION THREAD

**Discussion by kezara** [Mapsui/Mapsui#1811] [Q&A] [answered]
Hello good people,
I'm new to MapsUI and I tried to figure it out how to insert custom bitmaps from file on local disk, which is not embeded resource, to layer and after that to add that layer to Map layers for display on map, but I couldn't find the way how. I always end with BitmapId and rendering embeded resource file. I'm asking this as I see this as a way to allow end user to insert his own markers to map.

Is it possible to use not embeded resource files for map markers and how? If this is not possible, is there any other way to allow end user to insert his own markers to map? Thanks.

Edit:
I'm trying MAUI library

**Comment by pauldendulk** [comment, marked as answer]
You can load a stream, register it and use the bitmapId in the SymbolStyle.

```csharp
var stream = File.OpenRead("c:/image.png");
var bitmapId = BitmapRegistry.Instance.Register(stream);

symbolStyle = new SymbolStyle
{
    BitmapId = bitmapId,
    SymbolOffset = new RelativeOffset(0.0, 0.5),
    SymbolScale = 2
};
```

We never used files in our samples because they are platform independent projects which in the past did not support file access. We need to add a sample for that.

**Reply by kezara** [reply]
Sorry for late reply, this is what I needed. Many thanks.
